src/Microsoft.Azure.WebJobs.Extensions.ServiceBus/Triggers/UserTypeArgumentBindingProvider.cs

```diff
@@ -71,32 +72,46 @@ public async Task<ITriggerData> BindAsync(Message value, ValueBindingContext con
                 return new TriggerData(provider, bindingData);
             }
 
-            private static TInput GetBody(Message message, ValueBindingContext context)
+            private static TInput GetBody(Message message)
             {
+                // 1. If ContentType is "application/json" deserialize as JSON.
+                // 2. If ContentType is not "application/json" deserialize the body of a message that was serialized using XmlObjectSerializer.
+                // 3. If XmlObjectSerializer deserialization is failed deserialize as JSON.
+
                 if (message.ContentType == ContentTypes.ApplicationJson)
                 {
-                    string contents;
-
-                    contents = StrictEncodings.Utf8.GetString(message.Body);
-
+                    return DeserializeJsonObject(message);
+                }
+                else
+                {
                     try
                     {
-                        return JsonConvert.DeserializeObject<TInput>(contents, Constants.JsonSerializerSettings);
+                        return message.GetBody<TInput>();
```

Comment:
I don't understand what has changed here. Isn't this what the existing code does in this case already - call `message.GetBody<TInput>()`?

Reply:
In scenario "language worker" + "SB output" we send POCO json as "text/plain". Language worker execution result always treats as string. It requires more work if we want to change this behavior. 

C# SB output POCO -> C# SB Trigger POCO: works
Java SB output POCO -> C# SB Trigger POCO: does not work

Reply:
So the change you're making here is just an additional json deserialization attempt to cover cases where content type is specified incorrectly.

Reply:
> So the change you're making here is just an additional json deserialization attempt to cover cases where content type is specified incorrectly.

exactly